ca/certificate-authority.go

```diff
@@ -298,6 +315,31 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 		}
 	}
 
+	// Process requested extensions.  For now, the only extension we support is
+	// TLS Feature [RFC7633], i.e., "Must Staple", and for that we ignore the
+	// value provided by the client and just overwrite it with the specific value
+	// that only requires stapling.
+	//
+	// Other requested extensions are silently ignored.
+	requestedExtensions := []signer.Extension{}
+	for _, attr := range csr.Attributes {
+		if !attr.Type.Equal(oidExtensionRequest) {
+			continue
+		}
+
+		for _, extList := range attr.Value {
+			for _, ext := range extList {
+				if ext.Type.Equal(oidTLSFeature) {
```

Comment:
Indeed, any randomly requested TLS extension will result in a certificate with the `status_request` extension in the certificates `TLS Feature` extension, even though this was not explicitly requested. But @bifurcation already says so in the comment above :wink:  

Although I understand your concern, what are the odds someone requesting a `TLS Feature extension` without the intent for OCSP Must Staple (i.e. `status_request`)? The RFC mentions `status_request_v2` too, but I'm not that familiar with that extension to know if it's commonly used. (But any other extension could be requested in the CSR in theory..)

But if a explicit check for `mustStapleFeatureValue` is easily incorporated and wanted, I too would vote for diligence.